Ideas-2021.md

```diff
@@ -96,6 +97,79 @@ will help you to gain experience
 **Assisting Mentors**<br>
 To be added.
 
+--------------
+
+## Human Interface Guidelines Redesign for Sugar
+
+**Selected**<br>
+Nobody.
+
+**Prerequisites**<br>
+- Experience with UX design
+- Extensive experience with Python and GTK3
+- Experience with Sugar activities/toolkit
+- Experience with user interface and graphics design (and SVG)
+
+**Description**<br>
+Sugar’s use of color and icons (described in detail here [HIG
+color](https://wiki.sugarlabs.org/go/Human_Interface_Guidelines/The_Sugar_Interface/Colors)
+and [HIG
+icons](https://wiki.sugarlabs.org/go/Human_Interface_Guidelines/The_Sugar_Interface/Icons))
+is functional but a bit tired when compared to modern desktops and
+mobile systems.
+
+This project is about redesigning the use of color in Sugar in order
+to enable full-color icons for both the desktop itself and activities.
+This would require rewriting the [Human Interface Guidelines](https://wiki.sugarlabs.org/go/Human_Interface_Guidelines).
+
+The primary role of color in the current icon design is to;
+ - Indicate whether or not an activity has been used
+ - Indicate whether or not an activity is being shared with another
+   Sugar user (the colors of the person who launches the activity show
+   up on the desktops of the people who join the activity.)
+
+Fortunately, Sugar also support a mechanism for putting badges on
+icons. An example is in the neighborhood view, where badges are used
+to indicate which access points are active (See
+[networkviews.py](https://github.com/sugarlabs/sugar/blob/master/src/jarabe/desktop/networkviews.py)).
+
+We could use badges (See
+[icon.py](https://github.com/sugarlabs/sugar-toolkit-gtk3/blob/master/src/sugar3/graphics/icon.py#L49))
+to replace the functionality of color described above: for example, an
+XO badge to indicate an activity has been used. And the color of that
+badge could indicate collaboration. This would free up the icon itself
+to take on any colors deemed suitable by the activity designer.
+
+**Project Task:**<br>
+ - [ ] Rewrite the Human Interface Guidelines. Design a new idea for
+       the Sugar UX and UI.
+ - [ ] Add color functionality to the badges;
+ - [ ] Add badges in every instance where we currently use color: the
+       desktop, the journal, the neighborhood view and the activity
+       toolbar;
+ - [ ] Work with the design team to come up with new color icons for
+       all of the core Sugar toolbars, activities and activity toolbars
+       
+**Optional Tasks:**<br>
+ - [ ] Improve the Sugar Shell desktop UI, such as adding Animations 
+ - [ ] Port Sugar Toolkit to GTK4 (GTK4 has extensive support for animations)
```

Comment:
Can this optional task, be added as a separate project idea? I think it requires a greater time commitment and cannot be added as a task here.

Reply:
Agreed. 

Btw, Did we receive the number of task allocated to Sugar Labs?

Reply:
Done as of https://github.com/sugarlabs/GSoC/pull/114/commits/309366c3a69127889c93be3d12f1058506af673d

Reply:
Sir can i start working on the project

Reply:
We are a permissionless community; you don't need to ask for permission, you can start whatever you like.  As far as Sugar Labs is concerned, if you finish this before we start GSoC, we get what we want, and we won't need to run the project.  If you get a good start on it, and then make a proposal as a GSoC student, then that will demonstrate your capacity, and we will take it into account.  If there are others who also build on your work, they may achieve still greater things.  If you aren't even hoping to be part of GSoC, then that's fine too.  Sugar Labs gets what we need.  :grin: